### PR TITLE
editoast: lmr towed max speed

### LINKED ISSUE #9637: Towed rolling stocks with max speed at `0km/h` mess up the consist building
### What happened?

The consists are build with the formula:
`max_speed = min(max_speed_traction_engine, max_speed_towed)`

but when there's no data for the max speed towed the value is actually 0km/h, so the consist has a 0km/h value ⚙ 

### What did you expect to happen?

We should not take into account max_speed of towed rolling stocks if they are at 0km/h

### How can we reproduce it (as minimally and precisely as possible)?

.

### On which environments the bug occurs?

Recette (SNCF), Dev (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

b74fd59

### PR DESCRIPTION
**Pull request by Wadjetz**
fix [#9637](https://github.com/OpenRailAssociation/osrd/issues/9637)

- [x] Add stdcm request validation
- [x] Handle max speed zero value in simulation computation (if it used in another place without request validation)
- [x] Add max_speed to towed rolling stock

The frontend validation done in another PR

### CODE REVIEW

File: editoast/src/core/simulation.rs
Comment (woshilapin) [inline editoast/src/core/simulation.rs]: You can express that with a filter.

```rust
self.max_speed
  .filter(|max_speed_parameter| *max_speed_parameter != 0.0)
  .map(|max_speed_parameter| f64::min(self.traction_engine.max_speed, max_speed_parameter))
  .unwrap_or(self.traction_engine.max_speed)
```

File: editoast/src/views/timetable/stdcm.rs
Comment (woshilapin) [inline editoast/src/views/timetable/stdcm.rs]: Where does this `1.0` value comes from. I believe the ticket only mentions excluding `0.0`. You should be able to do that with the following.

```suggestion
    #[validate(range(exclusive_min = 0.0))]
```

File: editoast/src/views/timetable/stdcm.rs
Comment (woshilapin) [inline editoast/src/views/timetable/stdcm.rs]:
```suggestion
    #[validate(range(exclusive_min = 0.0))]
```

File: editoast/src/views/timetable/stdcm.rs
Comment (woshilapin) [inline editoast/src/views/timetable/stdcm.rs]:
```suggestion
    #[validate(range(exclusive_min = 0.0))]
```

**Review by woshilapin — COMMENTED** [review]
The code is fine (only some minor comments), but I'm questioning the original ticket https://github.com/OpenRailAssociation/osrd/issues/9637 in which I made a comment. I propose to not merge the current PR until we do have an answer.

**Review by hamz2a — APPROVED** [review]
LGTM, thanks!

**Review by flomonster — APPROVED** [review]
LGTM. Can you fixup your commits before merge.

**Review by woshilapin — CHANGES_REQUESTED** [review]
I don't think we should filter the zero, for the same reason I mentioned in https://github.com/OpenRailAssociation/osrd/issues/9637#issuecomment-2464324459. The fact that the field is optional should be enough.

**Review by woshilapin — APPROVED** [review]
Thank you.

**Comment by leovalais** [comment]
Ping @Tristramg: changes in rolling stock speed calculations, probably best to rebase before typing quantities any further :)

**Review by SharglutDev — APPROVED** [review]
Lgtm for front (not tested)

**Pull request merged**
